include/swift/ABI/TaskStatus.h

```diff
@@ -154,6 +159,37 @@ class ChildTaskStatusRecord : public TaskStatusRecord {
   }
 };
 
+/// A status record which states that a task has a task group, and potentially
+/// task group children inside it.
+class GroupChildTaskStatusRecord : public ChildTaskStatusRecord {
```

Comment:
This is stored in the parent, and is a way to iterate over all children of a group;

I'm not entirely super sure about the way it is used -- every child is a new record; should it rather by trying to add to the linked list a record maintains? There wasn't much prior art on how we want to store these child records so I wasnt sure